### PR TITLE
CE-3050 Recent changes across Wikia on Special:RecentChanges

### PR DESCRIPTION
**Pull request by adamkarminski**
Added a collapsible box with a message about recent changes across Wikia. Tracking added as well.

/cc: @Wikia/community-engineering

### CODE REVIEW

File: resources/mediawiki.special/mediawiki.special.recentchanges.js [hunk 1]
```diff
@@ -10,6 +10,58 @@
...
+					}
+				}
+			}
+
+			$legendElements.each( function () {
+				var $this = $(this),
+					id = $this.attr('id');
+
+				if (id !== null) {
+					if (!!cache.get(prefix + id)) {
```
Comment (lukaszkonieczny) [inline resources/mediawiki.special/mediawiki.special.recentchanges.js]: You can put this check to `if` above.

File: resources/mediawiki.special/mediawiki.special.recentchanges.js [hunk 2]
```diff
@@ -10,6 +10,58 @@
...
+						toggleCollapsible($this.parent());
+					}
+				}
+			});
+
+			$legendElements.on('click', function(e) {
```
Comment (lukaszkonieczny) [inline resources/mediawiki.special/mediawiki.special.recentchanges.js]: You can bind once, but while it is temporary code it's up to you ;)

File: includes/specials/SpecialRecentchanges.php [hunk 3]
```diff
@@ -670,6 +678,23 @@ function getExtraOptions( $opts ) {
...
+	function setRecentChangesOnWikia() {
+		$content = Html::rawElement( 'legend', [ 'id' => 'recentchanges-on-wikia' ],
+			wfMessage( 'recentchanges-on-wikia-title' )->escaped()
+		);
+		$content .= wfMessage( 'recentchanges-on-wikia-content' )->parse();
+
+		$this->getOutput()->addHTML(
+			Html::rawElement( 'fieldset',
+				[
+					'class' => 'collapsible collapsed',
```
Comment (lukaszkonieczny) [inline includes/specials/SpecialRecentchanges.php]: Maybe our box should be expanded as a initial state?

Comment (adamkarminski) [inline includes/specials/SpecialRecentchanges.php]: Actually, you're right! :+1:

File: resources/mediawiki.special/mediawiki.special.recentchanges.js (hunk 1 above)
Comment (adamkarminski) [inline resources/mediawiki.special/mediawiki.special.recentchanges.js]: :+1:

File: resources/mediawiki.special/mediawiki.special.recentchanges.js [hunk 4]
```diff
@@ -10,6 +10,64 @@
...
+
+			function toggleCollapsible($collapsible) {
+				$collapsible.toggleClass('collapsed');
+				updateCollapsedCache($collapsible);
+			}
+
+			function updateCollapsedCache($collapsible) {
+				var id = $collapsible.attr('id');
+
+				if (id !== null) {
```
Comment (lukaszkonieczny) [inline resources/mediawiki.special/mediawiki.special.recentchanges.js:25]: Please use this convention `id !== null` or `!!id``

File: resources/mediawiki.special/mediawiki.special.recentchanges.js [hunk 5]
```diff
@@ -10,6 +10,64 @@
...
+					if ($collapsible.hasClass('collapsed')) {
+						cache.set(prefix + id, 'collapsed', cache.CACHE_LONG);
+					} else {
+						cache.set(prefix + id, 'expanded', cache.CACHE_LONG);
```
Comment (lukaszkonieczny) [inline resources/mediawiki.special/mediawiki.special.recentchanges.js:29]: Do we need to set it while it's not used anywhere?

File: resources/mediawiki.special/mediawiki.special.recentchanges.js [hunk 6]
```diff
@@ -10,6 +10,64 @@
...
+			}
+
+			$collapsibleElements.each(function () {
+				var $this = $(this),
+					id = $this.attr('id');
+
+				if (!!id) {
+					var previousState = cache.get(prefix + id);
+
+					if (!!previousState) {
```
Comment (lukaszkonieczny) [inline resources/mediawiki.special/mediawiki.special.recentchanges.js:41]: Do we need this check?

**Comment by lukaszkonieczny** [comment]
After offline conversation LGTM! :+1:

File: resources/mediawiki.special/mediawiki.special.recentchanges.js (hunk 4 above)
Comment (adamkarminski) [inline resources/mediawiki.special/mediawiki.special.recentchanges.js:25]: `!==` used :)

**Pull request merged**